compiler/hash-vm/src/bytecode.rs

```diff
@@ -0,0 +1,283 @@
+    },
+
+    IDiv8 {
+        l1: Register,
+        l2: Register,
+    },
+    IMul8 {
+        l1: Register,
+        l2: Register,
+    },
+    IDiv16 {
+        l1: Register,
+        l2: Register,
+    },
+    IMul16 {
+        l1: Register,
+        l2: Register,
+    },
+    IDiv32 {
+        l1: Register,
+        l2: Register,
+    },
+    IMul32 {
+        l1: Register,
+        l2: Register,
+    },
+    IDiv64 {
+        l1: Register,
+        l2: Register,
+    },
+    IMul64 {
+        l1: Register,
+        l2: Register,
+    },
+    AddF32 {
+        l1: Register,
+        l2: Register,
+    },
+    SubF32 {
+        l1: Register,
+        l2: Register,
+    },
+    DivF32 {
+        l1: Register,
+        l2: Register,
+    },
+    MulF32 {
+        l1: Register,
+        l2: Register,
+    },
+    ModF32 {
+        l1: Register,
+        l2: Register,
+    },
+    AddF64 {
+        l1: Register,
+        l2: Register,
+    },
+    SubF64 {
+        l1: Register,
+        l2: Register,
+    },
+    DivF64 {
+        l1: Register,
+        l2: Register,
+    },
+    MulF64 {
+        l1: Register,
+        l2: Register,
+    },
+    ModF64 {
+        l1: Register,
+        l2: Register,
+    },
+    /// Logical Exclusive OR
+    Xor {
+        l1: Register,
+        l2: Register,
+    },
+    /// Logical OR
+    Or {
+        l1: Register,
+        l2: Register,
+    },
+    /// Logical AND (Like there exists a Non-Logical and)
+    And {
+        l1: Register,
+        l2: Register,
+    },
+    Not {
+        l1: Register,
+    },
+
+    /// We don't have integer variants as there is no need for them
+    PowF32 {
+        l1: Register,
+        l2: Register,
+    },
+    PowF64 {
+        l1: Register,
+        l2: Register,
+    },
+
+    Shl8 {
+        l1: Register,
+        l2: Register,
+    },
+    Shr8 {
+        l1: Register,
+        l2: Register,
+    },
+    Shl16 {
+        l1: Register,
+        l2: Register,
+    },
+    Shr16 {
+        l1: Register,
+        l2: Register,
+    },
+    Shl32 {
+        l1: Register,
+        l2: Register,
+    },
+    Shr32 {
+        l1: Register,
+        l2: Register,
+    },
+    Shl64 {
+        l1: Register,
+        l2: Register,
+    },
+    Shr64 {
+        l1: Register,
+        l2: Register,
+    },
+    /// Call a function at a given address?
+    Call {
+        func: Register,
+    },
+    /// Copy a value from source register to destination register.
+    Mov {
+        src: Register,
+        dest: Register,
+    },
+    Syscall {
+        id: Register,
+    },
+    Return,
+
+    /// The jump family (jumpin' around the christmas tree)
+    Jmp {
+        location: Register,
+    },
+    JmpPos {
+        l1: Register,
+        location: Register,
+    },
+    JmpNeg {
+        l1: Register,
+        location: Register,
+    },
+    JmpZero {
+        l1: Register,
+        location: Register,
+    },
+    // Compare both values and store the result in `l1`
```

Comment:
Maybe document that this returns -1 | 1 | 0

Reply:
This returns any range of values, the comparison should only jump if something is either zero, less than zero, or greater than zero.